File: 217-contains-duplicate/217-contains-duplicate.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> numsSet = new HashSet<>();
        for (int num : nums) {
            if (numsSet.contains(num)) {
                return true;
            }
            numsSet.add(num);
        }
        return false;
    }
}

File: 217-contains-duplicate/217-contains-duplicate.cpp
#include <unordered_set>
#include <vector>

class Solution {
 public:
  bool containsDuplicate(std::vector<int>& nums) {
    std::unordered_set<int> nums_set;
    for (int x : nums) {
      if (nums_set.count(x)) {
        return true;
      }
      nums_set.insert(x);
    }
    return false;
  }
};
